libp2p/crypto/ecnist.nim

```diff
@@ -663,34 +709,43 @@ proc init*[T: EcPKI](sospk: var T, data: string): Result[void, Asn1Error] {.inli
   ## Procedure returns ``Asn1Status``.
   sospk.init(fromHex(data))
 
-proc init*(t: typedesc[EcPrivateKey], data: openarray[byte]): EcPrivateKey =
+proc init*(t: typedesc[EcPrivateKey], data: openarray[byte]): EcResult[EcPrivateKey] =
```

Comment:
and these would be `fromDer` - not sure if it's feasible though - or is `DER` really the canonical binary format here?